src/datetime.rs

```diff
@@ -171,6 +171,16 @@ impl crate::DateTime {
         self.to_chrono()
             .to_rfc3339_opts(chrono::SecondsFormat::AutoSi, true)
     }
+
+    /// Convert the given RFC 3339 format string to a [`DateTime`].
```

Comment:
can we add a note about truncating to millisecond precision, similar to the one above in `from_chrono`?

Reply:
Added in https://github.com/mongodb/bson-rust/pull/317/commits/9496a4f83481f037935d44bbcbeeedbb882e383a.